Source/DafnyLanguageServer/Workspace/ChangeProcessors/Relocator.cs

```diff
@@ -99,6 +138,26 @@ Position afterChangeEndOffset
         return new Position(changeStart.Line + changeEof.Line, characterOffset);
       }
 
+      private static Range? MigrateRange(Range rangeToMigrate, Range changeRange, Position afterChangeEndOffset) {
+        if (!rangeToMigrate.Contains(changeRange) && rangeToMigrate.Intersects(changeRange)) {
+          // Do not migrate ranges that partially overlap with the change
+          return null;
```

Comment:
Don't you want to migrate one of the endpoints here?

Reply:
Well, I think if the entire range of a diagnostics has been removed, for example an ensures clause has been removed, then we shouldn't migrate the diagnostic. 

Similarly, if either the start or end of a diagnostic has been removed, I'm guessing it's better not to migrate it.

Reply:
>I'm not sure what you mean by "If a change occurs before and after a border of a diagnostic". Do you mean the text deleted by the change covers the entire range of the diagnostic? In that case this PR is not migrating the diagnostic, since it seems reasonable to guess that by deleting the offending code the diagnostic will disappear as well.

For example
```
^^^^^^^^^^^^^^^ (diagnostic)
          -------- (text removal)
^^^^^^^^^^ (diagnostic)
```

And in case the change removes the entire diagnostic, you should also delete it if it's not already what you are doing. Since these diagnostics are published and result in red squiggles, you don't want red squiggles in unrelated text.

Reply:
This PR will, if the change overlaps either the beginning or end of a diagnostic, delete instead of migrate the diagnostic.